addons/csw/functions/fnc_ai_reload.sqf

```diff
@@ -6,91 +6,68 @@
  * Arguments:
  * 0: CSW <OBJECT>
  * 1: Gunner <OBJECT>
- * 2: Weapon <STRING>
- * 3: Magazine <STRING> (default: "")
+ * 2: Skip reload time <BOOL> (default: false)
+ * 3: Clear forced magazine after reloading (default: false)
  *
  * Return Value:
  * None
  *
  * Public: No
  */
 
-params ["_vehicle", "_gunner", "_weapon", ["_magazine", ""]];
+params ["_vehicle", "_gunner", ["_instantReload", false], ["_clearForcedMag", false]];
+TRACE_3("AI reload",_vehicle,_gunner,_instantReload);
 
-private _turretPath = [_gunner] call EFUNC(common,getTurretIndex);
-private _reloadSource = objNull;
-private _reloadMag = "";
-private _reloadNeededAmmo = -1;
+// API, used for ai_switchMagazine
+private _forcedMag = _vehicle getVariable [QGVAR(forcedMag), ""];
+private _turretPath = _vehicle unitTurret _gunner;
+if (_turretPath isEqualTo []) then {
+    _turretPath = [0];
+};
 
-private _cfgMagGroups = configFile >> QGVAR(groups);
+// If this is called while CSW has ammo, unload mags in gunner's turret
+if (someAmmo _vehicle) then {[_vehicle, _turretPath] call FUNC(unloadMagazines)};
```

Comment:
Should we be unloading at this stage, when we don't know if we can even reload at all?

Reply:
Line 31 has an `exitWith`